### PR TITLE
Dm 2208

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2208

## Description - what does this code do?
1. General file cleanup.
2. Redirects user to `Instructions` page of new PE if they browse to an old editor page.
3. Reconfigures breadcrumbs.

## Testing done - how did you test it/steps on how can another person can test it 
1. Log in as an admin.
2. Try to visit the following editor endpoints: `/collaborators`, `/impact`, `/documentation`, `/resources`, `/departments`, `/timeline`, `/risk_and_mitigation`, and `/checklist`.
3. Make sure you are redirected to the current PE `Instructions` page for each one of the endpoints above.
4. If you did not create a `competitions/shark-tank` page via the page builder, try to visit that endpoint and make sure you are redirected to the homepage.
5. If you do have a page builder shark tank page, delete it and the `competitions` page group, and then browse to `/competitions/shark-tank` and make sure you're redirected to the homepage.
6. Go to the admin panel and create a page-builder page with a page group that does not have a landing page.
7. Browse to the newly created page. Make sure the page group breadcrumb text is not a link.
8. From the page, browse to a practice's show page either via the URL bar or from the practices list(if you added one).
9. Make sure the page group breadcrumb text is still not a link.
10. Repeat this process, but use a page group that has a landing page and make sure the breadcrumb text is a link on the page-builder page and the practice show page.
11. Visit `/diffusion-map` and make sure there are now two breadcrumbs at the top.

**_Breadcrumbs (edge cases)_**
1. Visit any practice's show page.
2. Using the URL bar, browse to a different practice's show page.
3. Make sure the current practice's name replaces the previous practice's name in the breadcrumbs section(this was to avoid having too many breadcrumbs at once).
4. Visit `/search`.
5. Without adding a query, use the URL bar to browse to any practice's show page.
6. Make sure there is a `Search` breadcrumb that goes to `/search`.

## Screenshots, Gifs, Videos from application (if applicable)
Screen Shot 2020-11-25 at 11 17 25 AM
Screen Shot 2020-11-25 at 11 19 56 AM
Screen Shot 2020-11-25 at 11 20 18 AM

## Acceptance criteria
- Remove views that we don't use
- Redirect them to the existing editor instead of old views
- Redirect to Editor Page home if it’s editor (Instructions)
- Redirect to homepage if it’s not an editor page
- Keep collaborators view

## Definition of done
- [X] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [X] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by bradjohnson92008 — DISMISSED** [review]
I was able to get through all the tests and verify all the functionality... looks like you nailed it!  Great work Josh!

File: app/helpers/navigation_helper.rb [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+    def practice_breadcrumb(practice)
+      session[:breadcrumbs].find { |b| b['display'] == practice.name }
+    end
+
+    def add_practice_breadcrumb(practice)
+      session[:breadcrumbs] << { 'practice': 'true', 'display': practice.name, 'path': practice_path(practice) }.stringify_keys
+    end
+
+    def add_checklist_breadcrumb(practice)
```
Comment (emyl3) [inline app/helpers/navigation_helper.rb:53]: this can be removed

File: app/helpers/navigation_helper.rb [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+          session[:breadcrumbs].delete_if { |bc| bc['practice'].present? }
+          add_practice_breadcrumb(practice_by_id)
+          # If there are any duplicate breadcrumbs, delete them
+        elsif practice_breadcrumb(practice_by_id).present? && practice_breadcrumb(practice_by_id).count > 1
+          remove_breadcrumb(practice_breadcrumb(practice_by_id))
+          add_practice_breadcrumb(practice_by_id)
+        end
+      end
+
+      # practice checklist path
```
Comment (emyl3) [inline app/helpers/navigation_helper.rb:104]: this also can be removed

**Review by emyl3 — CHANGES_REQUESTED** [review]
Looks good to me! Just a couple more things that can be removed! Also we can delete this file: `app/views/practices/planning_checklist.html.erb` too! Thanks Josh!

**Comment by JoshingYou1** [comment]
> Looks good to me! Just a couple more things that can be removed! Also we can delete this file: `app/views/practices/planning_checklist.html.erb` too! Thanks Josh!

I think I got everything!

**Review by emyl3 — APPROVED** [review]
Looks good to me! Thank you Josh!

**Pull request merged**